### PR TITLE
KOGITO-9561: Jobs service kafka and http messaging modularization

### PR DESCRIPTION
**Pull request by wmedvede**
Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [X] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [X] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [X ] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [X] Pull Request contains link to the JIRA issue
- [X] Pull Request contains link to any dependent or related Pull Request
- [X] Pull Request contains description of the issue
- [X] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [kogito-apps|kogito-examples] tests</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [kogito-apps|kogito-examples] quarkus-main</b>

- for <b>quarkus lts checks</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins run quarkus-lts</b>

- for a <b>specific quarkus lts check</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins (re)run [kogito-apps|kogito-examples] quarkus-lts</b>
 
- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [kogito-apps|kogito-examples] native</b>

- for <b>native lts checks</b>  
  Run native checks against quarkus lts branch
  Please add comment: <b>Jenkins run native-lts</b>

- for a <b>specific native lts check</b>  
  Run native checks against quarkus lts branch
  Please add comment: <b>Jenkins (re)run [kogito-apps|kogito-examples] native-lts</b>
 
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

<details>
<summary>
Quarkus-3 PR check is failing ... what to do ?
</summary>
The Quarkus 3 check is applying patches from the `.ci/environments/quarkus-3/patches`.

The first patch, called `0001_before_sh.patch`, is generated from Openrewrite `.ci/environments/quarkus-3/quarkus3.yml` recipe. The patch is created to speed up the check. But it may be that some changes in the PR broke this patch.  
No panic, there is an easy way to regenerate it. You just need to comment on the PR:
```
jenkins rewrite quarkus-3
```
and it should, after some minutes (~20/30min) apply a commit on the PR with the patch regenerated.

Other patches were generated manually. If any of it fails, you will need to manually update it... and push your changes.
</details>

### CODE REVIEW

**Comment by kie-ci3** [comment]
**(tests) - kogito-apps job** `#1222` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>


export BUILD_MVN_OPTS_CURRENT=-Dvalidate-formatting -Prun-code-coverage
build-chain build cross_pr -f 'https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml' -o 'bc' -p kiegroup/kogito-apps -u https://github.com/kiegroup/kogito-apps/pull/1805 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.kogito-apps/1222/display/redirect

**Test results:**
- PASSED: 2767
- FAILED: 5

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.kogito-apps/1222/testReport/org.kie.kogito.it.jobs/CallbackStateTimeoutsIT/Build_projects___callbackStateTimeoutsExceeded/">Build projects / org.kie.kogito.it.jobs.CallbackStateTimeoutsIT.callbackStateTimeoutsExceeded</a></summary>
java.util.concurrent.CompletionException: java.lang.RuntimeException: Unable to start Quarkus test resource class org.kie.kogito.test.resources.JobServiceCompositeQuarkusTestResource
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.kogito-apps/1222/testReport/org.kie.kogito.it.jobs/MultipleTimerInstancesIT/Build_projects___boundaryTimerEvent/">Build projects / org.kie.kogito.it.jobs.MultipleTimerInstancesIT.boundaryTimerEvent</a></summary>
java.util.concurrent.CompletionException: java.lang.RuntimeException: Unable to start Quarkus test resource class org.kie.kogito.test.resources.JobServiceCompositeQuarkusTestResource
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.kogito-apps/1222/testReport/org.kie.kogito.it.jobs/ProcessAsyncIT/Build_projects___testAsync/">Build projects / org.kie.kogito.it.jobs.ProcessAsyncIT.testAsync</a></summary>
java.util.concurrent.CompletionException: java.lang.RuntimeException: Unable to start Quarkus test resource class org.kie.kogito.test.resources.JobServiceCompositeQuarkusTestResource
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.kogito-apps/1222/testReport/org.kie.kogito.it.jobs/ProcessTimerIT/Build_projects___testDeleteTimerCycle/">Build projects / org.kie.kogito.it.jobs.ProcessTimerIT.testDeleteTimerCycle</a></summary>
java.util.concurrent.CompletionException: java.lang.RuntimeException: Unable to start Quarkus test resource class org.kie.kogito.test.resources.JobServiceCompositeQuarkusTestResource
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-apps.tests.kogito-apps/1222/testReport/org.kie.kogito.it.jobs/SwitchStateTimeoutsIT/Build_projects___switchStateEventConditionTimeoutsTransitionTimeoutsExceeded/">Build projects / org.kie.kogito.it.jobs.SwitchStateTimeoutsIT.switchStateEventConditionTimeoutsTransitionTimeoutsExceeded</a></summary>
java.util.concurrent.CompletionException: java.lang.RuntimeException: Unable to start Quarkus test resource class org.kie.kogito.test.resources.JobServiceCompositeQuarkusTestResource
</details>

File: jobs-service/jobs-service-messaging-http/src/main/resources/META-INF/microprofile-config.properties [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+# limitations under the License.
+#
+
+# This file contains the default configurations for the Quarkus http connector.
+
+# Job status change events notification disabled by default
+kogito.jobs-service.http.job-status-change-events=false
+
+# Job status change events are automatically enabled with the profile: http-events-support (-Dquarkus.profile=http-events-support)
+%http-events-support.kogito.jobs-service.http.job-status-change-events=true
```
Comment (tiagodolphine) [inline jobs-service/jobs-service-messaging-http/src/main/resources/META-INF/microprofile-config.properties:23]: do we really need the profile? I mean can't we just set the config property `kogito.jobs-service.http.job-status-change-events` to enable it?

File: jobs-service/jobs-service-messaging-kafka/src/main/resources/META-INF/microprofile-config.properties [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This file contains the default configurations for Kafka.
+
+# Kafka global default configurations.
+%kafka-events-support.kafka.bootstrap.servers=localhost:9092
```
Comment (tiagodolphine) [inline jobs-service/jobs-service-messaging-kafka/src/main/resources/META-INF/microprofile-config.properties]: Just thinking but can't we just get rid of the profile? I think is is needed to disable during runtime right?
We can use an application properties with a suffix for the profile just to make it cleaner, https://quarkus.io/guides/config-reference#profile-aware-files

**Review by tiagodolphine — COMMENTED** [review]
LGTM, I left some comments but my concern is with the profiles and profile name changes, like `events-support` changing this is a breaking change, mainly with probably BDD tests, operator, etc where we were injecting this profile. 🤔

File: jobs-service/jobs-service-messaging-http/src/main/resources/META-INF/microprofile-config.properties (hunk 1 above)
Comment (wmedvede) [reply jobs-service/jobs-service-messaging-http/src/main/resources/META-INF/microprofile-config.properties:23]: yes, we can remove. It was added to have let's say some degree of consistency with the kafka-events-support profile. But not being used too much.

Comment (wmedvede) [reply jobs-service/jobs-service-messaging-http/src/main/resources/META-INF/microprofile-config.properties:23]: it also remind us to keep consistency with the data-index, that now has the kafka-events-support and http-events-support profiles.

File: jobs-service/jobs-service-messaging-kafka/src/main/resources/META-INF/microprofile-config.properties (hunk 2 above)
Comment (wmedvede) [reply jobs-service/jobs-service-messaging-kafka/src/main/resources/META-INF/microprofile-config.properties]: yeah, I know this possibility of having the profile-aware-files, and in fact, it's something we use in the examples.
But, for the server modules, honestly I don't like this profile-aware-file approach, and is not aligned with data-index, etc.
I think this is ok, also now, if you look with the refactoring the configurations starts to look much clearer, etc.

**Comment by wmedvede** [comment]
> 

This is a good point, the thing is, now we have the http based events and the kafka based events.
So, the question is, do we keep this adhered to the past, or we do something for moving forward?
Is it good to have the world "kafka" when you are configuring kafka?
I think that yes, but yeah, It's something to evaluate.

By doing a quick inspection, I'll re-review tomorrow, I believe that the potential change is not too big.
I mean, kogito-apps and kogito-runtime integration tests, etc, are working well.
Then, we have the "operator", "the image" and the "documentaion", I need to double check, but IDK if the "operator" uses this value, because for tuning of the eventing API, (when we introduced kafka events to create jobs, they added a config map in the deployment produced by the "operator" to let people add whatever config they want, and IDK if the "operator" is doing too much with this "events-support" value. Need to check.

**Comment by wmedvede** [comment]
> LGTM, I left some comments but my concern is with the profiles and profile name changes, like `events-support` changing this is a breaking change, mainly with probably BDD tests, operator, etc where we were injecting this profile. thinking

This is a good point, the thing is, now we have the http based events and the kafka based events.
So, the question is, do we keep this adhered to the past, or we do something for moving forward?
Is it good to have the world "kafka" when you are configuring kafka?
I think that yes, but yeah, It's something to evaluate.

By doing a quick inspection, I'll re-review tomorrow, I believe that the potential change is not too big.
I mean, kogito-apps and kogito-runtime integration tests, etc, are working well.
Then, we have the "operator", "the image" and the "documentaion", I need to double check, but IDK if the "operator" uses this value, because for tuning of the eventing API, (when we introduced kafka events to create jobs, they added a config map in the deployment produced by the "operator" to let people add whatever config they want, and IDK if the "operator" is doing too much with this "events-support" value. Need to check.

"image" : a potential change I think is not too much.

"documentation": to change the old kogito documentation is a couple of lines.

File: jobs-service/jobs-service-messaging-kafka/src/main/resources/META-INF/microprofile-config.properties [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This file contains the default configurations for Kafka.
+
+# Kafka global default configurations.
+%events-support,kafka-events-support.kafka.bootstrap.servers=localhost:9092
```
Comment (wmedvede) [inline jobs-service/jobs-service-messaging-kafka/src/main/resources/META-INF/microprofile-config.properties:20]: @tiagodolphine , See here, I think we can keep the events-support for backward compatibility, that will work the same as the kafka-events-support, but, in the SW documentation we talk about kafka-events-support.

wdyt?

Comment (tiagodolphine) [reply jobs-service/jobs-service-messaging-kafka/src/main/resources/META-INF/microprofile-config.properties:20]: nice I didn't know that it is possible to declare more than one profile to a property
👍

**Review by tiagodolphine — APPROVED** [review]

**Pull request merged**